src/librustc_target/spec/mod.rs

```diff
@@ -676,6 +676,7 @@ supported_targets! {
     ("powerpc64-wrs-vxworks", powerpc64_wrs_vxworks),
 
     ("mipsel-sony-psp", mipsel_sony_psp),
+    ("thumbv4t-nintendo-gba", thumbv4t_nintendo_gba),
```

Comment:
The CPU name is a bit odd as the trailing T means thumb support, but I guess this makes sense. You *could* run ARM instructions on the GBA though, right?

We do support one other ARMv4 target, `armv4t-unknown-linux-gnueabi`, but I believe that defaults to ARM instructions.

Reply:
Yeah both a32 and t32 are supported, and the system boots in a32 state, but we want as much of the code as possible to be compiled using t32, so we're defaulting to t32. With the [instruction set RFC](https://github.com/rust-lang/rfcs/pull/2867) it'll be a lot easier to mix the two, but we'll still want the default to be t32 for any untagged functions.

Reply:
Oh but I see what you mean with the trailing `t` since the name already starts with `thumb`.

i guess it could just be `thumbv4-nintendo-gba`

Reply:
Oh I remember now!! it's called `thumbv4t-nintendo-gba` because the LLVM target string that LLVM uses as the "basis" for this target profile is `thumbv4t-none-eabi`, and so if they put "thumbv4t" then we should kinda just follow suit.

I remember that somehow I was able to run a command line thing and have LLVM or rustc print a list of all the "known" target strings, but I can't remember how to do that.

Reply:
You mean `rustc --print target-list`?

Reply:
Sorta like that, but I thought that I'd seen a similar command called on LLVM directly. I could be imagining that part.